### PR TITLE
[JENKINS-45892] IllegalStateException at hudson.XmlFile.replaceIfNotA…

### PR DESCRIPTION
**Pull request by carlossg**
…tTopLevel

[JENKINS-45892](https://issues.jenkins-ci.org/browse/JENKINS-45892) introduced a change that is causing

```
Nov 29, 2017 6:58:52 PM hudson.XmlFile replaceIfNotAtTopLevel
WARNING: JENKINS-45892: reference to org.jenkinsci.plugins.workflow.job.WorkflowJob@488f6bf6[cloudbees/test/branch] being saved from unexpected /var/jenkins_home/jobs/test/branches/branch.njvm0k/builds/7/build.xml
java.lang.IllegalStateException
        at hudson.XmlFile.replaceIfNotAtTopLevel(XmlFile.java:210)
```

### CODE REVIEW

**Comment by carlossg** [comment]
@reviewbybees

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractInvisibleRunAction2.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ */
+public abstract class AbstractInvisibleRunAction2 extends InvisibleAction implements RunAction2 {
+
+    private static final Logger LOGGER = Logger.getLogger(AbstractInvisibleRunAction2.class.getName());
+
+    protected final Stack<String> stack = new Stack<>();
+
+    private transient Run<?, ?> run;
+
+    public AbstractInvisibleRunAction2(Run<?, ?> run) {
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractInvisibleRunAction2.java]: As I understand `RunAction2`, this is not necessary, as `onAttached(Run)` will be called as soon as it is attached to a `Run` instance.

Comment (carlossg) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/AbstractInvisibleRunAction2.java]: it's used in the tests https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java#L209

**Comment by carlossg** [comment]
@reviewbybees done

**Pull request merged**